### ISSUE TITLE
Qgis Ubuntu Repository Guidelines not followed

### ISSUE THREAD

**Issue by thetuxinator** [qgis/QGIS#50747]
### What is the bug or the crash?

Hi all,

Some time ago i discovered that the qgis ubuntu/debian repositories don't follow the best-practice.

lets begin with
https://ubuntu.qgis.org/ubuntu/dists/focal/main/binary-amd64/Packages

QGIS-Package file above only contains the most recent version of qgis-server which makes it impossible to downgrade to an older (even minor only) Version of Qgis or to stay on a particular version of a package till you decide yourself if you want to upgrade.

Example:

The file at the Moment contains the following:

Package: qgis-server
Source: qgis
Version: 1:3.22.12+32focal

However, if we would have all the current minor-versions available like this:

pool/main/q/qgis/qgis-server_3.22.12+32focal_amd64.deb
pool/main/q/qgis/qgis-server_3.22.11+32focal_amd64.deb
pool/main/q/qgis/qgis-server_3.22.9+32focal_amd64.deb
pool/main/q/qgis/qgis-server_3.22.8+32focal_amd64.deb
pool/main/q/qgis/qgis-server_3.22.7+32focal_amd64.deb
pool/main/q/qgis/qgis-server_3.22.5+32focal_amd64.deb

in https://ubuntu.qgis.org/ubuntu/dists/focal/main/binary-amd64/Packages

which would look like 

Package: qgis-server
Source: qgis
Version: 1:3.22.12+32focal

Package: qgis-server
Source: qgis
Version: 1:3.22.11+32focal

Package: qgis-server
Source: qgis
Version: 1:3.22.9+32focal

Package: qgis-server
Source: qgis
Version: 1:3.22.8+32focal

Package: qgis-server
Source: qgis
Version: 1:3.22.7+32focal

Package: qgis-server
Source: qgis
Version: 1:3.22.5+32focal

Doing apt install would install latest version 3.22.12+32focal by default, however doing "apt search qgis-server" would result in an info message similar to "more versions available, use -a to show all versions" .
This gives the ability to provide an exact version which you want to install, which allows selective up and downgrades. an example apt command looks like apt install qgis-server=3.22.11+32focal

This would make testing and staying on a minor version much easier for everybody. also it would follow ubuntu's and debians Package Format Guidelines.

Another error i found in the repository format is, you use ubuntu-ltr as the "distribution" name. Thats wrong as of the Format Guidelines https://wiki.debian.org/DebianRepository/Format

While Qgis-Repository's at the moment look like:

https://ubuntu.qgis.org/ubuntu-ltr/

https://ubuntu.qgis.org/ubuntu/

they should better look like:

https://ubuntu.qgis.org/ubuntu/dists/focal/main/
https://ubuntu.qgis.org/ubuntu/dists/focal/ltr/

or 
https://ubuntu.qgis.org/ubuntu/dists/focal/qgis/
https://ubuntu.qgis.org/ubuntu/dists/focal/qgis-ltr/

Or you select lts/non-lts by its package-name and have them in the same path repo like https://ubuntu.qgis.org/ubuntu/dists/focal/qgis/ for example

qgis-server
qgis-server-lts

where qgis-server-lts could even be a virtual package if you follow aboves versioning principle and keep every version (also majors in that case) in the package file.

Kind Regards

Mike

### Steps to reproduce the issue

Add the Qgis Ubuntu Repositories

### Versions

All Deb/Ubuntu Packages

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

**Comment by thetuxinator** [comment]
A good example of how it can be done with having all package major versions in a single "main" repo can be found at https://apt.postgresql.org/pub/repos/apt/dists/focal-pgdg/main/binary-amd64/

**Comment by thetuxinator** [comment]
Hi, as we have to use some workarounds (mirroring/snapshoting repo by script) to force using a particular version on PROD, i would like to ask if there is any progress on this?

**Comment by m-kuhn** [comment]
If understand correctly there has been an update on this topic and future versions will be properly added, is this correct @jef-n ?

**Comment by jef-n** [comment]
> If understand correctly there has been an update on this topic and future versions will be properly added, is this correct @jef-n ?

Yes, reprepro didn't allow multiple versions before version 5.4.  That's apparently still in experimental, but we're using that now and have multiple versions (5 for the each nightly/distribution; unlimited for releases).   Oddly the latest version of 5.4 seems to be broken by a fix, that introduces an issue for me that has a lot of similarity to the apparently rarely occuring issue, that it strives to fix - w/o that it apparently works.

And now I gotta see why the a build ran out of space - I guess I have a hunch ;)
